apps/browser-extension-wallet/src/hooks/useWalletManager.ts

```diff
@@ -528,6 +574,78 @@ export const useWalletManager = (): UseWalletManager => {
     [getCurrentChainId]
   );
 
+  const createSharedWallet = async ({
+    mnemonic,
```

Comment:
The mnemonic will be the same as in `ownSignerWalletId`:
```ts
    const wallets = await firstValueFrom(walletRepository.wallets$);
    const ownWallet = wallets.find(({walletId}) => walletId === ownSignerWalletId)
    // you can create the key agent with this (I think in a constructor), instead of from mnemonic
    ownWallet.encryptedSecrets.rootPrivateKeyBytes
```

Also best to pass account index as an argument to a utility function at this level
```suggestion
    accountIndex
```

Reply:
Do you mean we should create keyAgent with [this](https://github.com/input-output-hk/cardano-js-sdk/blob/master/packages/key-management/test/InMemoryKeyAgent.test.ts#L259-L278) approach instead to avoid exposing the mnemonic?

Also, on `ownWallet.encryptedSecrets.rootPrivateKeyBytes`, I get this error
```
Property encryptedSecrets does not exist on type AnyWallet<WalletMetadata, AccountMetadata>
```